source/parser/function.civet

```diff
@@ -670,6 +671,8 @@ function iterationDeclaration(statement: IterationStatement | ForStatement)
         when "max" then "-Infinity"
         when "product" then "1"
         else "0"
+  else if object
+    declaration.children.push "={}"
```

Comment:
This `if`/`else if` gives priority to `reduction` over `object` here, whereas below you give priority to `object`. I think these should be consistent. Maybe `object` should be given priority, given the `...` that precedes it.

(You could also do error detection (`reduction and object`) here, instead of in `parser.hera`.)